roles/openshift_master/tasks/main.yml

```diff
@@ -164,6 +164,22 @@
     - restart master api
     - restart master controllers
 
+- name: Check for credentials file for registry auth
+  stat:
+    path: "{{oreg_auth_credentials_path }}"
+  when:
+    - oreg_auth_user is defined
+  register: master_oreg_auth_credentials_stat
+
+- name: Create credentials for registry auth
+  command: "docker --config={{ oreg_auth_credentials_path }} login -u {{ oreg_auth_user }} -p {{ oreg_auth_password }} {{ oreg_host }}"
+  when:
+    - oreg_auth_user is defined
+    - (not master_oreg_auth_credentials_stat.stat.exists or oreg_auth_credentials_replace) | bool
+  notify:
+    - restart master api
+    - restart master controllers
```

Comment:
Unfortunately for 3.6 and older we need to make sure to notify `restart master` for non HA installs and these two for HA installs.